acorn/src/options.js

```diff
@@ -121,7 +121,7 @@ export function getOptions(opts) {
   if (options.allowReserved == null)
     options.allowReserved = options.ecmaVersion < 5
 
-  if (options.allowHashBang == null)
+  if (options.allowHashBang === false || options.allowHashBang == null)
```

Comment:
I'd say that explicitly setting this to false should be possible, even for `ecmaVersion >= 14`.

Reply:
currently the problem when using `lastest ecma (2023)` and no other options, `hashbang` is by default turned off, which is probably not intended.

https://github.com/acornjs/acorn/blob/ae85582ae0a9f3397acf9497f1dcc4c9811b95da/acorn/src/options.js#L50

just curious, what is the practical reason to be able to turn it off? my impression was that this option was only added because a lot of code used the hashbang grammar prior to when it was officially adopted into the spec. that said, I would apply the same argument to `allowAwaitOutsideFunction`.

Reply:
ref: https://github.com/acornjs/acorn/issues/1134

Reply:
if you have a really strong opinion about this then we might have to move the condition to the applicable code, similar to what has been done with `allowAwaitOutsideFunction`, e.g. https://github.com/acornjs/acorn/blob/ae85582ae0a9f3397acf9497f1dcc4c9811b95da/acorn/src/state.js#L112

Reply:
Now that the option exists, I feel that ignoring it when explicitly set is going to be unexpected to users. I don't see how this could end being turned off when not given with `ecmaVersion="latests"`. `options.versions` is initialized to a high number above, so if this is null, it will be set to true.

Reply:
only glanced over he code, but I think since the [default option is false](https://github.com/acornjs/acorn/blob/ae85582ae0a9f3397acf9497f1dcc4c9811b95da/acorn/src/options.js#L50), it remains false when not set.

Reply:
I see what you mean—it'll get set to `false` in the loop earlier. Patch e708a22 should help with this.

Reply:
for some reason the code has been like his before: https://github.com/acornjs/acorn/commit/b2ecf7a378097a95c9244209cd5e4aff748c2b2c but was changed to fix[?] something else https://github.com/acornjs/acorn/issues/1134

Reply:
Good point. See 11340a4

Reply:
I changed he PR title.